Update phpstan to 0.12

adaamz opened the pull request.

adaamz committed:
update phpstan

adaamz committed:
install & run phpstan in CI

mabar commented on .travis.yml:
```diff
@@ -10,23 +10,9 @@ before_install:
-    - stage: Quality Assurance
-      php: 7.2
-      script:
-        - composer run-script qa
```
Why are you changing build?

adaamz replied:
bcs master is failing and phpstan does not run in pipeline

this pr is still wip

f3l1x commented:
Could you please @adaamz repair CI?

adaamz committed:
fixes

adaamz commented:
I fixed the phpstan issues, but there is some inconsistence between PHP 7.2 and PHP 7.3 BcMath. Can anybody look on that? Thanks.

MartkCz commented:
Failing test fixed

@adaamz Could you fix cs, please?

adaamz commented:
@MartkCz 
- rebased
- fixed issues
- updated to phpstan 0.12
- installed phpstan/extension-installer (to really use required extensions in phpstan)
- removed phpstan-strict-rules (almost all rules where voleted)

f3l1x commented:
LGTM

f3l1x commented:
Merged. Thx.
